### PR TITLE
fix(web): headless recorder unhandled paths, invalid test-sequence references 🔩

### PR DESCRIPTION
**Pull request by jahorton**
This bugfix was originally part of #11424 and is now being spun off as its own PR.  It appears that previously, keyboard-simulation test failures would not have been reported properly if they occurred.  

The bug being fixed was discovered due to either "noImplicitAny" or "noImplicitThis" being enforced against the sourcefile.

I ended up extracting all changes from that PR, meaning that this PR makes the source file compatible with stricter .tsconfig settings.  I also added extra tweaks while I was at it, largely around type-names to facilitate checks against `'key' | 'osk'` so that TS could better make inferences about related control flow in line with "noImplicitReturns".

As this PR only adjusts unit-test resources, there's no need to bother 🍒-picking this.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/recorder/src/index.ts [hunk 1]
```diff
@@ -568,8 +571,8 @@ export class EventSpecTestSet implements TestSet<InputEventSpecSequence> {
     let testSet = this.testSet;
 
     for(var i=0; i < testSet.length; i++) {
-      var testSeq = this[i];
-      var simResult = await testSet[i].test(proctor);
+      const testSeq = this.testSet[i];
+      const simResult = await testSet[i].test(proctor);
```
Comment (mcdurdin) [inline common/web/recorder/src/index.ts:575]:
```suggestion
      const testSeq = this.testSet[i];
      const simResult = await testSeq.test(proctor);
```

tis but a nit

File: common/web/recorder/src/index.ts [hunk 2]
```diff
@@ -618,8 +621,8 @@ export class RecordedSequenceTestSet implements TestSet<RecordedKeystrokeSequenc
...
```
Comment (mcdurdin) [inline common/web/recorder/src/index.ts:625]:
```suggestion
      const testSeq = this.testSet[i];
      const simResult = await testSeq.test(proctor);
```

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Pull request merged**